plutus-contract/test/Spec/Contract/Tx/Constraints/MustSpendAtLeast.hs

```diff
@@ -61,51 +68,53 @@ mustSpendAtLeastContract offAmt onAmt = do
             Constraints.collectFromTheScript utxos ()
             <> Constraints.mustIncludeDatumInTx (Datum $ PlutusTx.toBuiltinData onAmt)
             <> Constraints.mustSpendAtLeast (Ada.lovelaceValueOf offAmt)
-    ledgerTx2 <- submitTxConstraintsWith @UnitTest lookups2 tx2
+    ledgerTx2 <- submitTxFromConstraints lookups2 tx2
     awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx2
 
 trace :: Contract () Empty ContractError () -> Trace.EmulatorTrace ()
 trace contract = do
     void $ Trace.activateContractWallet w1 contract
     void Trace.nextSlot
 
-entireScriptBalance :: TestTree
-entireScriptBalance =
-    let contract = mustSpendAtLeastContract scriptBalance scriptBalance
+entireScriptBalance :: SubmitTx -> TestTree
+entireScriptBalance sub =
+    let contract = mustSpendAtLeastContract sub scriptBalance scriptBalance
     in  checkPredicateOptions
             defaultCheckOptions
             "Successful use of mustSpendAtLeast at script's exact balance"
             (assertValidatedTransactionCount 2)
             (void $ trace contract)
 
-lessThanScriptBalance :: TestTree
-lessThanScriptBalance =
+lessThanScriptBalance :: SubmitTx -> TestTree
+lessThanScriptBalance sub =
     let amt = scriptBalance - 1
-        contract = mustSpendAtLeastContract amt amt
+        contract = mustSpendAtLeastContract sub amt amt
     in  checkPredicateOptions
             defaultCheckOptions
             "Successful use of mustSpendAtLeast below script's balance"
             (assertValidatedTransactionCount 2)
             (void $ trace contract )
 
-higherThanScriptBalance :: TestTree
-higherThanScriptBalance =
+higherThanScriptBalance :: SubmitTx -> TestTree
+higherThanScriptBalance sub =
     let amt = scriptBalance + 5_000_000
-        contract = mustSpendAtLeastContract amt amt
+        contract = mustSpendAtLeastContract sub amt amt
     in  checkPredicateOptions
             defaultCheckOptions
             "Validation pass when mustSpendAtLeast is greater than script's balance and wallet's pubkey is included in the lookup"
             (assertContractError contract (Trace.walletInstanceTag w1)
-              (has $ _ConstraintResolutionContractError . _DeclaredInputMismatch)
-              "failed to throw error"
+                (\case
+                    { ConstraintResolutionContractError (Constraints.DeclaredInputMismatch _) -> True
+                    ; TxConstraintResolutionContractError (Tx.Constraints.LedgerMkTxError (Constraints.DeclaredInputMismatch _)) -> True
```

Comment:
(connected to previous comment) Ideally, we should be able to check the same error in both cases here.